research/conditional/moe_layers/expert_choice.py

```diff
@@ -21,8 +21,10 @@ def __init__(
         expert_size: int,
         topk_fraction: float,
         random_perm: bool = False,
+        group_by_batch: bool = False,
+        one_hot_impl: bool = False,
+        softmax_ungrouped: bool = False,
```

Comment:
wolałbym "softmax_grouped", bez negacji. Bo później mamy podwójną negację przy not_softmax_ungrouped, i ciężko się czyta.

Reply:
to jest flaga na inputcie, one domyślnie są na false w konwencji całego naszego repo. I chyba jest to w miare czytelne, bo naturalnie wszystko jest "False", a "True" to dodatkowy feature włączony.